pulp/apis/scip_api.py

```diff
@@ -108,35 +113,38 @@ def actualSolve(self, lp):
     def readsol(filename):
         """Read a SCIP solution file"""
         with open(filename) as f:
-            # First line must containt 'solution status: <something>'
+            values = {}
+
+            # First line must contain 'solution status: <something>'
             try:
                 line = f.readline()
                 comps = line.split(': ')
                 assert comps[0] == 'solution status'
                 assert len(comps) == 2
-            except:
-                raise PulpSolverError("Can't read SCIP solver output: %r" % line)
+            except Exception:
+                raise PulpSolverError("Can't get SCIP solver status: %r" % line)
 
             status = SCIP_CMD.SCIP_STATUSES.get(comps[1].strip(), constants.LpStatusUndefined)
 
-            # Look for an objective value. If we can't find one, stop.
-            try:
-                line = f.readline()
-                comps = line.split(': ')
-                assert comps[0] == 'objective value'
-                assert len(comps) == 2
-                float(comps[1].strip())
-            except:
-                raise PulpSolverError("Can't read SCIP solver output: %r" % line)
+            if status not in SCIP_CMD.NO_SOLUTION_STATUSES:
```

Comment:
maybe better to flip this to  and exit early:

```python
if status in SCIP_CMD.NO_SOLUTION_STATUSES:
    return status, values
```

Reply:
Nicer I agree, will change!